src/gui/filedetails/shareemodel.cpp

```diff
@@ -233,9 +261,48 @@ void ShareeModel::shareesFetched(const QJsonDocument &reply)
 
     beginResetModel();
     _sharees = newSharees;
+    insertSearchGloballyItem(newSharees);
     endResetModel();
 
     Q_EMIT shareesReady();
+
+    setLookupMode(LookupMode::LocalSearch);
+}
+
+void ShareeModel::insertSearchGloballyItem(const QVector<ShareePtr> &newShareesFetched)
+{
+    const auto isAlreadyInserted = std::find_if(std::begin(_sharees), std::end(_sharees), [](const ShareePtr &sharee) {
+        return sharee->type() == Sharee::LookupServerSearch || sharee->type() == Sharee::LookupServerSearchResults;
+    }) != std::end(_sharees);
+
+    if (!isAlreadyInserted) {
+        _sharees.push_back(_searchGloballyPlaceholder);
+    }
+
+    if (lookupMode() == LookupMode::GlobalSearch) {
+        const auto displayName = newShareesFetched.isEmpty() ? tr("No results found") : tr("Global search results");
+        _searchGloballyPlaceholder->setDisplayName(displayName);
+        _searchGloballyPlaceholder->setType(Sharee::LookupServerSearchResults);
+    } else {
+        _searchGloballyPlaceholder->setDisplayName(tr("Search globally"));
+        _searchGloballyPlaceholder->setType(Sharee::LookupServerSearch);
+    }
+
+    sortSharees();
+}
+
+void ShareeModel::sortSharees()
+{
+    // sort _sharees such that lookup placeholder is always last
+    std::sort(std::begin(_sharees), std::end(_sharees), [](const auto &left, const auto &right) {
+        const auto isRightElementLookupPlaceholder = right->type() == Sharee::LookupServerSearch
+            || right->type() == Sharee::LookupServerSearchResults;
+
+        const auto isLeftElementLookupPlaceholder = left->type() == Sharee::LookupServerSearch
+            || left->type() == Sharee::LookupServerSearchResults;
+
+        return !isLeftElementLookupPlaceholder && isRightElementLookupPlaceholder;
+    });
 }
```

Comment:
If we need to sort the elements in the model this should be done in a separate `QSortFilterProxyModel` (with testing for the correct order), and through an override of `QSortFilterProxyModel::lessThan`